src/main.js

```diff
@@ -175,28 +168,30 @@ function makePoster(){
 }
 
 function saveMainPoster() {
-  currentPoster = new Poster(posterImg.src, posterTitle.innerHTML, posterQuote.innerHTML);
-    for(var i = 0; i < savedPosters.length; i++ ) {
-      if (savedPosters[i].url == currentPoster.url &&
-        savedPosters[i].title == currentPoster.title &&
-        savedPosters[i].quote == currentPoster.quote) {
-        var existingPoster = savedPosters[i]
-      }
-    }
-    if (!existingPoster) {
-      savedPosters.push(currentPoster);
-      makePoster();
-    }
+  if (!existingPoster()) {
+    savedPosters.push(currentPoster);
+    makePoster();
+  }
+}
+
+function existingPoster(){
+  return savedPosters.find((poster) => {
+    poster.imageURL === currentPoster.imageURL &&
+    poster.title === currentPoster.title &&
+    poster.quote === currentPoster.quote
+  })
 }
 
 function loadRandomPoster() {
```

Comment:
Can we please touch base on the updates to the functions loadRandomPoster() and generatePoster() ? Thank you!